nginxQuic: 3be953161026 -> 0af598651e33

Izorkin opened the pull request:
###### Description of changes
Update nginxQuic to revision a954b551dc3f.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Izorkin committed:
nginxQuic: 3be953161026 -> 0af598651e33

Izorkin commented:
@GrahamcOfBorg test nginx

Izorkin commented:
cc @ajs124

ajs124 commented:
Any idea whats up with https://logs.nix.ci/?key=nixos/nixpkgs.207248&attempt_id=4efa516b-5c45-43c3-9670-8a89908b23f3 ?

Izorkin commented:
On my system it helped to increase the sleep time before retesting etag.
```diff
diff --git a/nixos/tests/nginx.nix b/nixos/tests/nginx.nix
index 73f1133bd6c..ead8a644c3b 100644
--- a/nixos/tests/nginx.nix
+++ b/nixos/tests/nginx.nix
@@ -95,7 +95,7 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         webserver.succeed(
             "${etagSystem}/bin/switch-to-configuration test >&2"
         )
-        webserver.sleep(1)
+        webserver.sleep(5)
         new_etag = check_etag()
         assert old_etag != new_etag
```
I tried to make these changes here, but it didn't work.
Maybe have to rewrite this test...

ajs124 commented:
Hm. Maybe we can loop?

Izorkin commented:
How to do it? Right now I'm out of ideas.

Izorkin commented:
Update to revision 5fd628b89bb7.

Izorkin committed:
nixos/nginx: update quic configuration

Izorkin commented:
Update QUIC configuration. Fixed warning message:
```
nginx[59371]: nginx: [warn] the "http3" parameter is deprecated, use "quic" parameter instead in /etc/nginx/nginx.conf:587
```

RaitoBezarius commented:
Shortlog so far:
```
2 days ago | Sergey Kandaurov | HTTP/3: fixed OpenSSL compatibility layer initialization. quic tip
-- | -- | --
3 days ago | Maxim Dounin | SSL: enabled TLSv1.3 by default. default
3 days ago | Maxim Dounin | Mail: fixed handling of blocked client read events in proxy.
2 weeks ago | Maxim Dounin | Syslog: introduced error log handler.
2 weeks ago | Maxim Dounin | Syslog: removed usage of ngx_cycle->log and ngx_cycle->hostname.
2 weeks ago | Maxim Dounin | HTTP/2: finalize request as bad if header validation fails.
2 weeks ago | Maxim Dounin | HTTP/2: socket leak with "return 444" in error_page (ticket #2455).
2 weeks ago | Maxim Dounin | SSL: logging levels of errors observed with BoringSSL.
2 weeks ago | Maxim Dounin | SSL: logging levels of errors observed with tlsfuzzer and LibreSSL.
2 weeks ago | Maxim Dounin | SSL: logging levels of various errors reported with tlsfuzzer.
2 weeks ago | Maxim Dounin | SSL: switched to detect log level based on the last error.
4 weeks ago | Yugo Horie | Core: stricter UTF-8 handling in ngx_utf8_decode().
2 months ago | Roman Arutyunyan | HTTP/3: trigger more compatibility errors for "listen quic". quic
4 weeks ago | Roman Arutyunyan | HTTP/3: "quic" parameter of "listen" directive. quic
4 weeks ago | Roman Arutyunyan | QUIC: OpenSSL compatibility layer. quic
4 weeks ago | Maxim Dounin | Win32: non-ASCII names in ngx_fs_bsize(), ngx_fs_available().
4 weeks ago | Maxim Dounin | Win32: removed attempt to use a drive letter in ngx_fs_bsize().
4 weeks ago | Maxim Dounin | Win32: non-ASCII names support in ngx_open_tempfile().
4 weeks ago | Maxim Dounin | Win32: non-ASCII names support in ngx_rename_file().
4 weeks ago | Maxim Dounin | Win32: non-ASCII names support in ngx_delete_file().
4 weeks ago | Maxim Dounin | Win32: reworked ngx_win32_rename_file() to use nginx wrappers.
4 weeks ago | Maxim Dounin | Win32: reworked ngx_win32_rename_file() to check errors.
4 weeks ago | Maxim Dounin | Win32: non-ASCII directory names support in ngx_delete_dir().
4 weeks ago | Maxim Dounin | Win32: non-ASCII directory names support in ngx_create_dir().
4 weeks ago | Maxim Dounin | Win32: non-ASCII directory names support in ngx_getcwd().
4 weeks ago | Maxim Dounin | Win32: non-ASCII names support in "include" with wildcards.
4 weeks ago | Maxim Dounin | Win32: non-ASCII names support in autoindex (ticket #458).
4 weeks ago | Maxim Dounin | Win32: OpenSSL compilation for x64 targets with MSVC.
4 weeks ago | Maxim Dounin | Win32: i386 now assumed when crossbuilding (ticket #2416).
4 weeks ago | Maxim Dounin | Win32: handling of localized MSVC cl output.
4 weeks ago | Maxim Dounin | Win32: removed unneeded wildcard in NGX_CC_NAME test for msvc.
4 weeks ago | Sergey Kandaurov | QUIC: improved ssl_reject_handshake error logging. quic
4 weeks ago | Sergey Kandaurov | QUIC: using ngx_ssl_handshake_log(). quic
4 weeks ago | Sergey Kandaurov | QUIC: moved "handshake failed" reason to send_alert. quic
4 weeks ago | Sergey Kandaurov | QUIC: using NGX_QUIC_ERR_CRYPTO macro in ALPN checks. quic
6 weeks ago | Sergey Kandaurov | QUIC: fixed indentation. quic
6 weeks ago | Sergey Kandaurov | README: fixed toc. quic
6 weeks ago | Sergey Kandaurov | README: updated building from sources, added directives reference. quic
7 weeks ago | Maxim Dounin | Lingering close for connections with pipelined requests.
8 weeks ago | Maxim Dounin | Fixed "zero size buf" alerts with subrequests.
8 weeks ago | Maxim Dounin | Style.
7 weeks ago | Roman Arutyunyan | QUIC: fixed broken token in NEW_TOKEN (ticket #2446). quic
7 weeks ago | Roman Arutyunyan | QUIC: ngx_quic_copy_buffer() function. quic
8 weeks ago | Maxim Dounin | Added warning about redefinition of listen socket protocol options.
2 months ago | Maxim Dounin | Fixed handling of very long locations (ticket #2435).
2 months ago | Maxim Dounin | Gzip static: ranges support (ticket #2349).
2 months ago | Maxim Dounin | Configure: removed unneeded header from UDP_SEGMENT test.
2 months ago | Maxim Dounin | QUIC: improved SO_COOKIE configure test. quic
2 months ago | Sergey Kandaurov | QUIC: defer setting the active flag for client stream events. quic
2 months ago | Roman Arutyunyan | QUIC: relocated ngx_quic_init_streams() for 0-RTT. quic
2 months ago | Roman Arutyunyan | HTTP/3: insert count block timeout. quic
2 months ago | Roman Arutyunyan | HTTP/3: trigger 400 (Bad Request) on stream error while blocked. quic
2 months ago | Roman Arutyunyan | QUIC: set stream error flag on reset. quic
2 months ago | Roman Arutyunyan | QUIC: automatically add and never delete stream events. quic
2 months ago | Sergey Kandaurov | HTTP/3: fixed $connection_time. quic
2 months ago | Roman Arutyunyan | HTTP/3: handled insertion reference to a going to be evicted entry. quic
2 months ago | Sergey Kandaurov | Merged with the default branch. quic
3 months ago | Maxim Dounin | Updated link to OpenVZ suspend/resume bug.
3 months ago | Valentin Bartenev | Fixed port ranges support in the listen directive.
3 months ago | BullerDu | Style.
3 months ago | Sergey Kandaurov | Version bump.
3 months ago | Sergey Kandaurov | Merged with the default branch. quic
3 months ago | Maxim Dounin | release-1.23.3 tag
3 months ago | Maxim Dounin | nginx-1.23.3-RELEASE release-1.23.3
3 months ago | Maxim Dounin | Updated OpenSSL and zlib used for win32 builds.
3 months ago | Maxim Dounin | Win32: event flags handling edge cases in ngx_wsarecv().
3 months ago | Maxim Dounin | SSL: fixed ngx_ssl_recv() to reset c->read->ready after errors.
3 months ago | Maxim Dounin | Removed casts from ngx_memcmp() macro.
3 months ago | Maxim Dounin | Fixed alignment of ngx_memmove()/ngx_movemem() macro definitions.
5 months ago | Roman Arutyunyan | HTTP/3: implement keepalive for hq. quic
3 months ago | Roman Arutyunyan | QUIC: application init() callback. quic
7 months ago | Roman Arutyunyan | HTTP/3: renamed functions. quic
3 months ago | Roman Arutyunyan | QUIC: removed cancelable flag from QUIC and HTTP/3 events. quic
5 months ago | Roman Arutyunyan | QUIC: idle mode for main connection. quic
5 months ago | Roman Arutyunyan | HTTP/3: unified hq code with regular HTTP/3 code. quic
6 months ago | Roman Arutyunyan | QUIC: do not send MAX_STREAMS in shutdown state. quic
7 months ago | Roman Arutyunyan | QUIC: defer stream removal until all its data is acked. quic
3 months ago | Roman Arutyunyan | QUIC: reusable mode for main connection. quic
6 months ago | Roman Arutyunyan | QUIC: post close event for connection close. quic
7 months ago | Roman Arutyunyan | QUIC: made ngx_quic_finalize_connecion() more graceful. quic
6 months ago | Roman Arutyunyan | QUIC: treat qc->error == -1 as a missing error. quic
4 months ago | Sergey Kandaurov | QUIC: fixed computation of nonce with packet numbers beyond 2^32. quic
4 months ago | Jiuzhou Cui | HTTP/3: fixed build without NGX_PCRE (broken by 0f5fc7a320db). quic
4 months ago | Sergey Kandaurov | SSL: fixed debug logging of SSL_sendfile() return value.
4 months ago | Maxim Dounin | Fixed segfault when switching off master process during upgrade.
4 months ago | Maxim Dounin | Disabled cloning of sockets without master process (ticket #2403).
4 months ago | Maxim Dounin | Filtering duplicate addresses in listen (ticket #2400).
4 months ago | Roman Arutyunyan | QUIC: fixed triggering stream read event (ticket #2409). quic
4 months ago | Roman Arutyunyan | Process events posted by ngx_close_idle_connections() immediately.
4 months ago | Sergey Kandaurov | QUIC: fixed C4334 MSVC warning about 32 to 64 bits conversion. quic
4 months ago | Sergey Kandaurov | QUIC: plug MSVC warning about potentially uninitialized variable. quic
4 months ago | Sergey Kandaurov | Added shutdown macros for win32 required for QUIC. quic
4 months ago | Sergey Kandaurov | QUIC: fixed C4389 MSVC warning about signed/unsigned mismatch. quic
4 months ago | Sergey Kandaurov | QUIC: avoid using C99 designated initializers. quic
4 months ago | Sergey Kandaurov | QUIC: moved variable declaration to fix build with MSVC 2010. quic
4 months ago | Sergey Kandaurov | QUIC: fixed C4706 warnings with MSVC 2010. quic
4 months ago | Sergey Kandaurov | HTTP/3: fixed server_name regex captures (ticket #2407). quic
4 months ago | Ciel Zhao | SSI: handling of subrequests from other modules (ticket #1263).
4 months ago | Maxim Dounin | Fixed PROXY protocol to use ngx_memcpy()/ngx_memcmp().
4 months ago | Maxim Dounin | Added logging to PROXY protocol write buffer check.
4 months ago | Roman Arutyunyan | Set default listen socket type in http. quic
4 months ago | Roman Arutyunyan | Increased maximum read PROXY protocol header size.
4 months ago | Roman Arutyunyan | Version bump.

2 days ago	Sergey Kandaurov	[HTTP/3: fixed OpenSSL compatibility layer initialization.](https://hg.nginx.org/nginx-quic/rev/5fd628b89bb7) quic tip
3 days ago	Maxim Dounin	[SSL: enabled TLSv1.3 by default.](https://hg.nginx.org/nginx-quic/rev/d1cf09451ae8) default
3 days ago	Maxim Dounin	[Mail: fixed handling of blocked client read events in proxy.](https://hg.nginx.org/nginx-quic/rev/a5e6e8510634)
2 weeks ago	Maxim Dounin	[Syslog: introduced error log handler.](https://hg.nginx.org/nginx-quic/rev/8771d35d55d0)
2 weeks ago	Maxim Dounin	[Syslog: removed usage of ngx_cycle->log and ngx_cycle->hostname.](https://hg.nginx.org/nginx-quic/rev/29adacffdefa)
2 weeks ago	Maxim Dounin	[HTTP/2: finalize request as bad if header validation fails.](https://hg.nginx.org/nginx-quic/rev/9db24374123b)
2 weeks ago	Maxim Dounin	[HTTP/2: socket leak with "return 444" in error_page (ticket #2455).](https://hg.nginx.org/nginx-quic/rev/72627f1cf09e)
2 weeks ago	Maxim Dounin	[SSL: logging levels of errors observed with BoringSSL.](https://hg.nginx.org/nginx-quic/rev/b7d4bfd132d2)
2 weeks ago	Maxim Dounin	[SSL: logging levels of errors observed with tlsfuzzer and LibreSSL.](https://hg.nginx.org/nginx-quic/rev/64db9e50f6c5)
2 weeks ago	Maxim Dounin	[SSL: logging levels of various errors reported with tlsfuzzer.](https://hg.nginx.org/nginx-quic/rev/6bee5e692579)
2 weeks ago	Maxim Dounin	[SSL: switched to detect log level based on the last error.](https://hg.nginx.org/nginx-quic/rev/69c7df4fe5d3)
4 weeks ago	Yugo Horie	[Core: stricter UTF-8 handling in ngx_utf8_decode().](https://hg.nginx.org/nginx-quic/rev/a10210a45c8b)
2 months ago	Roman Arutyunyan	[HTTP/3: trigger more compatibility errors for "listen quic".](https://hg.nginx.org/nginx-quic/rev/f4279edda9fd) quic
4 weeks ago	Roman Arutyunyan	[HTTP/3: "quic" parameter of "listen" directive.](https://hg.nginx.org/nginx-quic/rev/c851a2ed5ce8) quic
4 weeks ago	Roman Arutyunyan	[QUIC: OpenSSL compatibility layer.](https://hg.nginx.org/nginx-quic/rev/7da4791e0264) quic
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names in ngx_fs_bsize(), ngx_fs_available().](https://hg.nginx.org/nginx-quic/rev/2acb00b9b5ff)
4 weeks ago	Maxim Dounin	[Win32: removed attempt to use a drive letter in ngx_fs_bsize().](https://hg.nginx.org/nginx-quic/rev/c33eb93f9c7a)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names support in ngx_open_tempfile().](https://hg.nginx.org/nginx-quic/rev/e818ed227735)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names support in ngx_rename_file().](https://hg.nginx.org/nginx-quic/rev/96d894b38667)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names support in ngx_delete_file().](https://hg.nginx.org/nginx-quic/rev/4b06186670ed)
4 weeks ago	Maxim Dounin	[Win32: reworked ngx_win32_rename_file() to use nginx wrappers.](https://hg.nginx.org/nginx-quic/rev/37a184966ab3)
4 weeks ago	Maxim Dounin	[Win32: reworked ngx_win32_rename_file() to check errors.](https://hg.nginx.org/nginx-quic/rev/8880fe0b193c)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII directory names support in ngx_delete_dir().](https://hg.nginx.org/nginx-quic/rev/7d60e4584d9e)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII directory names support in ngx_create_dir().](https://hg.nginx.org/nginx-quic/rev/e0f385521c79)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII directory names support in ngx_getcwd().](https://hg.nginx.org/nginx-quic/rev/8ea2e052feb4)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names support in "include" with wildcards.](https://hg.nginx.org/nginx-quic/rev/751f79bd802c)
4 weeks ago	Maxim Dounin	[Win32: non-ASCII names support in autoindex (ticket #458).](https://hg.nginx.org/nginx-quic/rev/b0a06c50c1b4)
4 weeks ago	Maxim Dounin	[Win32: OpenSSL compilation for x64 targets with MSVC.](https://hg.nginx.org/nginx-quic/rev/3c4d81ea1338)
4 weeks ago	Maxim Dounin	[Win32: i386 now assumed when crossbuilding (ticket #2416).](https://hg.nginx.org/nginx-quic/rev/79c04253bc43)
4 weeks ago	Maxim Dounin	[Win32: handling of localized MSVC cl output.](https://hg.nginx.org/nginx-quic/rev/17b3efb45b17)
4 weeks ago	Maxim Dounin	[Win32: removed unneeded wildcard in NGX_CC_NAME test for msvc.](https://hg.nginx.org/nginx-quic/rev/9bcc5cc94ff4)
4 weeks ago	Sergey Kandaurov	[QUIC: improved ssl_reject_handshake error logging.](https://hg.nginx.org/nginx-quic/rev/639fa6723700) quic
4 weeks ago	Sergey Kandaurov	[QUIC: using ngx_ssl_handshake_log().](https://hg.nginx.org/nginx-quic/rev/0f4f781e57c1) quic
4 weeks ago	Sergey Kandaurov	[QUIC: moved "handshake failed" reason to send_alert.](https://hg.nginx.org/nginx-quic/rev/cb7dc35ed428) quic
4 weeks ago	Sergey Kandaurov	[QUIC: using NGX_QUIC_ERR_CRYPTO macro in ALPN checks.](https://hg.nginx.org/nginx-quic/rev/5dcea9f91482) quic
6 weeks ago	Sergey Kandaurov	[QUIC: fixed indentation.](https://hg.nginx.org/nginx-quic/rev/12b756caaf16) quic
6 weeks ago	Sergey Kandaurov	[README: fixed toc.](https://hg.nginx.org/nginx-quic/rev/93876a4c3b27) quic
6 weeks ago	Sergey Kandaurov	[README: updated building from sources, added directives reference.](https://hg.nginx.org/nginx-quic/rev/3c33d39a51d3) quic
7 weeks ago	Maxim Dounin	[Lingering close for connections with pipelined requests.](https://hg.nginx.org/nginx-quic/rev/cffaf3f2eec8)
8 weeks ago	Maxim Dounin	[Fixed "zero size buf" alerts with subrequests.](https://hg.nginx.org/nginx-quic/rev/f5515e727656)
8 weeks ago	Maxim Dounin	[Style.](https://hg.nginx.org/nginx-quic/rev/1c3b78d7cdc9)
7 weeks ago	Roman Arutyunyan	[QUIC: fixed broken token in NEW_TOKEN (ticket #2446).](https://hg.nginx.org/nginx-quic/rev/def8e398d7c5) quic
7 weeks ago	Roman Arutyunyan	[QUIC: ngx_quic_copy_buffer() function.](https://hg.nginx.org/nginx-quic/rev/3c98fa8fef6f) quic
8 weeks ago	Maxim Dounin	[Added warning about redefinition of listen socket protocol options.](https://hg.nginx.org/nginx-quic/rev/106328a70f4e)
2 months ago	Maxim Dounin	[Fixed handling of very long locations (ticket #2435).](https://hg.nginx.org/nginx-quic/rev/4eb1383f6432)
2 months ago	Maxim Dounin	[Gzip static: ranges support (ticket #2349).](https://hg.nginx.org/nginx-quic/rev/c7e103acb409)
2 months ago	Maxim Dounin	[Configure: removed unneeded header from UDP_SEGMENT test.](https://hg.nginx.org/nginx-quic/rev/ec26874c3f60)
2 months ago	Maxim Dounin	[QUIC: improved SO_COOKIE configure test.](https://hg.nginx.org/nginx-quic/rev/a954b551dc3f) quic
2 months ago	Sergey Kandaurov	[QUIC: defer setting the active flag for client stream events.](https://hg.nginx.org/nginx-quic/rev/03897c45798e) quic
2 months ago	Roman Arutyunyan	[QUIC: relocated ngx_quic_init_streams() for 0-RTT.](https://hg.nginx.org/nginx-quic/rev/bf2267887014) quic
2 months ago	Roman Arutyunyan	[HTTP/3: insert count block timeout.](https://hg.nginx.org/nginx-quic/rev/6bb884dc7291) quic
2 months ago	Roman Arutyunyan	[HTTP/3: trigger 400 (Bad Request) on stream error while blocked.](https://hg.nginx.org/nginx-quic/rev/eaa8dc3788e1) quic
2 months ago	Roman Arutyunyan	[QUIC: set stream error flag on reset.](https://hg.nginx.org/nginx-quic/rev/2ee6f9729739) quic
2 months ago	Roman Arutyunyan	[QUIC: automatically add and never delete stream events.](https://hg.nginx.org/nginx-quic/rev/35e27117b593) quic
2 months ago	Sergey Kandaurov	[HTTP/3: fixed $connection_time.](https://hg.nginx.org/nginx-quic/rev/e3760b9b7c8e) quic
2 months ago	Roman Arutyunyan	[HTTP/3: handled insertion reference to a going to be evicted entry.](https://hg.nginx.org/nginx-quic/rev/987bee4363d1) quic
2 months ago	Sergey Kandaurov	[Merged with the default branch.](https://hg.nginx.org/nginx-quic/rev/af5adec171b4) quic
3 months ago	Maxim Dounin	[Updated link to OpenVZ suspend/resume bug.](https://hg.nginx.org/nginx-quic/rev/07b0bee87f32)
3 months ago	Valentin Bartenev	[Fixed port ranges support in the listen directive.](https://hg.nginx.org/nginx-quic/rev/2af1287d2da7)
3 months ago	BullerDu	[Style.](https://hg.nginx.org/nginx-quic/rev/3108d4d668e4)
3 months ago	Sergey Kandaurov	[Version bump.](https://hg.nginx.org/nginx-quic/rev/d85ce1df2313)
3 months ago	Sergey Kandaurov	[Merged with the default branch.](https://hg.nginx.org/nginx-quic/rev/91ad1abfb285) quic
3 months ago	Maxim Dounin	[release-1.23.3 tag](https://hg.nginx.org/nginx-quic/rev/c38588d8376b)
3 months ago	Maxim Dounin	[nginx-1.23.3-RELEASE](https://hg.nginx.org/nginx-quic/rev/ff3afd1ce6a6) release-1.23.3
3 months ago	Maxim Dounin	[Updated OpenSSL and zlib used for win32 builds.](https://hg.nginx.org/nginx-quic/rev/9ed5778f5d4a)
3 months ago	Maxim Dounin	[Win32: event flags handling edge cases in ngx_wsarecv().](https://hg.nginx.org/nginx-quic/rev/56819a9491fe)
3 months ago	Maxim Dounin	[SSL: fixed ngx_ssl_recv() to reset c->read->ready after errors.](https://hg.nginx.org/nginx-quic/rev/06c7d84cafdb)
3 months ago	Maxim Dounin	[Removed casts from ngx_memcmp() macro.](https://hg.nginx.org/nginx-quic/rev/2ffefe2f892e)
3 months ago	Maxim Dounin	[Fixed alignment of ngx_memmove()/ngx_movemem() macro definitions.](https://hg.nginx.org/nginx-quic/rev/351d7f4e326f)
5 months ago	Roman Arutyunyan	[HTTP/3: implement keepalive for hq.](https://hg.nginx.org/nginx-quic/rev/b87a0dbc1150) quic
3 months ago	Roman Arutyunyan	[QUIC: application init() callback.](https://hg.nginx.org/nginx-quic/rev/b0c2234aaa9f) quic
7 months ago	Roman Arutyunyan	[HTTP/3: renamed functions.](https://hg.nginx.org/nginx-quic/rev/7b83da3bdf9f) quic
3 months ago	Roman Arutyunyan	[QUIC: removed cancelable flag from QUIC and HTTP/3 events.](https://hg.nginx.org/nginx-quic/rev/be39ffdf9208) quic
5 months ago	Roman Arutyunyan	[QUIC: idle mode for main connection.](https://hg.nginx.org/nginx-quic/rev/1192923be0aa) quic
5 months ago	Roman Arutyunyan	[HTTP/3: unified hq code with regular HTTP/3 code.](https://hg.nginx.org/nginx-quic/rev/6546c2ae1c7b) quic
6 months ago	Roman Arutyunyan	[QUIC: do not send MAX_STREAMS in shutdown state.](https://hg.nginx.org/nginx-quic/rev/740d7d6e8ff0) quic
7 months ago	Roman Arutyunyan	[QUIC: defer stream removal until all its data is acked.](https://hg.nginx.org/nginx-quic/rev/2e51cf3ffd90) quic
3 months ago	Roman Arutyunyan	[QUIC: reusable mode for main connection.](https://hg.nginx.org/nginx-quic/rev/37d5dddabaea) quic
6 months ago	Roman Arutyunyan	[QUIC: post close event for connection close.](https://hg.nginx.org/nginx-quic/rev/aaca8e111959) quic
7 months ago	Roman Arutyunyan	[QUIC: made ngx_quic_finalize_connecion() more graceful.](https://hg.nginx.org/nginx-quic/rev/2b4891fa0fbc) quic
6 months ago	Roman Arutyunyan	[QUIC: treat qc->error == -1 as a missing error.](https://hg.nginx.org/nginx-quic/rev/9c427e98d8c1) quic
4 months ago	Sergey Kandaurov	[QUIC: fixed computation of nonce with packet numbers beyond 2^32.](https://hg.nginx.org/nginx-quic/rev/70ce1e927715) quic
4 months ago	Jiuzhou Cui	[HTTP/3: fixed build without NGX_PCRE (broken by 0f5fc7a320db).](https://hg.nginx.org/nginx-quic/rev/210ad79a8853) quic
4 months ago	Sergey Kandaurov	[SSL: fixed debug logging of SSL_sendfile() return value.](https://hg.nginx.org/nginx-quic/rev/0b360747c74e)
4 months ago	Maxim Dounin	[Fixed segfault when switching off master process during upgrade.](https://hg.nginx.org/nginx-quic/rev/8852f39311de)
4 months ago	Maxim Dounin	[Disabled cloning of sockets without master process (ticket #2403).](https://hg.nginx.org/nginx-quic/rev/09463dd9c504)
4 months ago	Maxim Dounin	[Filtering duplicate addresses in listen (ticket #2400).](https://hg.nginx.org/nginx-quic/rev/4cc2bfeff46c)
4 months ago	Roman Arutyunyan	[QUIC: fixed triggering stream read event (ticket #2409).](https://hg.nginx.org/nginx-quic/rev/c6580dce98a8) quic
4 months ago	Roman Arutyunyan	[Process events posted by ngx_close_idle_connections() immediately.](https://hg.nginx.org/nginx-quic/rev/b809f53d3f5b)
4 months ago	Sergey Kandaurov	[QUIC: fixed C4334 MSVC warning about 32 to 64 bits conversion.](https://hg.nginx.org/nginx-quic/rev/2b3a8d5795c9) quic
4 months ago	Sergey Kandaurov	[QUIC: plug MSVC warning about potentially uninitialized variable.](https://hg.nginx.org/nginx-quic/rev/5b49f8bac1b4) quic
4 months ago	Sergey Kandaurov	[Added shutdown macros for win32 required for QUIC.](https://hg.nginx.org/nginx-quic/rev/96bdda15413a) quic
4 months ago	Sergey Kandaurov	[QUIC: fixed C4389 MSVC warning about signed/unsigned mismatch.](https://hg.nginx.org/nginx-quic/rev/e23fd55e1cc6) quic
4 months ago	Sergey Kandaurov	[QUIC: avoid using C99 designated initializers.](https://hg.nginx.org/nginx-quic/rev/8c0bccdf2743) quic
4 months ago	Sergey Kandaurov	[QUIC: moved variable declaration to fix build with MSVC 2010.](https://hg.nginx.org/nginx-quic/rev/a6cc246654f8) quic
4 months ago	Sergey Kandaurov	[QUIC: fixed C4706 warnings with MSVC 2010.](https://hg.nginx.org/nginx-quic/rev/a26897674420) quic
4 months ago	Sergey Kandaurov	[HTTP/3: fixed server_name regex captures (ticket #2407).](https://hg.nginx.org/nginx-quic/rev/0f5fc7a320db) quic
4 months ago	Ciel Zhao	[SSI: handling of subrequests from other modules (ticket #1263).](https://hg.nginx.org/nginx-quic/rev/49e7db44b57c)
4 months ago	Maxim Dounin	[Fixed PROXY protocol to use ngx_memcpy()/ngx_memcmp().](https://hg.nginx.org/nginx-quic/rev/42bc158a47ec)
4 months ago	Maxim Dounin	[Added logging to PROXY protocol write buffer check.](https://hg.nginx.org/nginx-quic/rev/251daa98cc87)
4 months ago	Roman Arutyunyan	[Set default listen socket type in http.](https://hg.nginx.org/nginx-quic/rev/6cf8ed15fd00) quic
4 months ago	Roman Arutyunyan	[Increased maximum read PROXY protocol header size.](https://hg.nginx.org/nginx-quic/rev/17d6a537fb1b)
4 months ago	Roman Arutyunyan	[Version bump.](https://hg.nginx.org/nginx-quic/rev/a2924855f453)
```

RaitoBezarius commented:
```
mars 27 13:50:19 media nginx-pre-start[24643]: nginx: [emerg] "listen" directive "proxy_protocol" parameter is incompatible with "quic" in /nix/store/9cajajyz8ygn4z2knhzg77wzp3bmv0gc-nginx>
```

This breaks PROXY protocol. Uncertain if it was already like this before (?).

RaitoBezarius commented:
https://hg.nginx.org/nginx-quic/rev/f4279edda9fd introduced this
Hmmmmm, are we still unable to run QUIC and HTTP/2 side by side?

RaitoBezarius commented:
> > Hmmmmm, are we still unable to run QUIC and HTTP/2 side by side?
> 
> Example:
> 
> ```
>   server {
>     listen 0.0.0.0:443 quic reuseport ;
>     listen 0.0.0.0:443 http2 ssl reuseport ;
> ...
>   server {
>     listen 0.0.0.0:443 quic ;
>     listen 0.0.0.0:443 http2 ssl ;
> ```

I think our NixOS module needs to be updated to filter out automatically the incompatible clauses as per my last message so that we don't break HTTP3 users configuration (like me :D): https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-servers/nginx/default.nix#L313-L329

RaitoBezarius commented:
Also, we should probably also rename http3 into quic too and deprecate http3 option.

Izorkin commented:
> I think our NixOS module needs to be updated to filter out automatically the incompatible clauses as per my last message so that we don't break HTTP3 users configuration (like me :D):

Should this be implemented in this PR?

Izorkin commented:
> This breaks PROXY protocol. Uncertain if it was already like this before (?).

```
          (if ssl && vhost.http3 then "
          # UDP listener for **QUIC+HTTP/3
          listen ${addr}:${toString port} http3 "
          + optionalString vhost.default "default_server "
          + optionalString vhost.reuseport "reuseport "
          + optionalString proxyProtocol "proxy_protocol "
```
You just need to delete this line in the http3 block:
```
          + optionalString proxyProtocol "proxy_protocol "
```

Izorkin commented:
> Also, we should probably also rename http3 into quic too and deprecate http3 option.

Changed logic work of the `http3` parameter and added the `quic` parameter.

RaitoBezarius commented on nixos/modules/services/web-servers/nginx/default.nix:
```diff
@@ -311,9 +311,9 @@ let
+          listen ${addr}:${toString port} quic "
           + optionalString vhost.default "default_server "
           + optionalString vhost.reuseport "reuseport "
           + optionalString (extraParameters != []) (concatStringsSep " " extraParameters)
```
So, taking the commit in hg, I see:
```
Now "ssl", "proxy_protocol" and "http2" are not allowed with "quic" in "listen"
directive.  Previously, only "ssl" was not allowed.
```

So, you could go like this:

```
let incompatibleParameters = [ "ssl" "proxy_protocol" "http2" ];
     isCompatibleParameter = param: !(any (p: p == param) incompatibleParameters);
in filter isCompatibleParameter extraParameters
```

(I let you adapt it to the exact code.)

Therefore, it's impossible to get a broken configuration especially if we generate the listen string ourselves.

Also, we probably want to have assertions errors for cases where the user explicitly set a "proxy_protocol"-only listener or "http2"-only listener, but that's probably out of scope for this PR as it needs to move those parameters in the signature and remove them from `extraParameters`.

RaitoBezarius replied:
(At least, the filter is needed because we could generate impossible NGINX configurations.)

Izorkin replied:
Hmm, could probably delete the line. 
```
+ optionalString (extraParameters != []) (concatStringsSep " " extraParameters)
```

RaitoBezarius replied:
I mean, I don't know what are other parameters compatible with QUIC, but we should probably keep it as it would constrain uselessly the QUIC configuration.
But we should filter out all **known incompatible** parameters.

Izorkin replied:
Resolved.

SuperSandro2000 commented on nixos/doc/manual/release-notes/rl-2305.section.md:
```diff
@@ -187,6 +187,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - [services.xserver.videoDrivers](options.html#opt-services.xserver.videoDrivers) now defaults to the `modesetting` driver over device-specific ones. The `radeon`, `amdgpu` and `nouveau` drivers are still available, but effectively unmaintained and not recommended for use.
 
+- Now, to enable the HTTP3 protocol in nginx, need to activate the parameter `services.nginx.virtualHosts.<<name>>.quic = true;` which is responsible for activating the QUIC transport protocol.
```
```suggestion
- To enable the HTTP3 (QUIC) protocol for a nginx virtual host, set the `quic` attribute on it to true, e.g. `services.nginx.virtualHosts.<name>.quic = true;`.
```

Izorkin replied:
Fixed.

SuperSandro2000 commented on nixos/modules/services/web-servers/nginx/vhost-options.nix:
```diff
@@ -199,13 +199,43 @@ with lib;
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable HTTP/0.9 protocol negotiation used in QUIC interoperability tests.
```
http3_hq and http/0.9?

Izorkin replied:
Yes, it says so in the Readme - http://hg.nginx.org/nginx-quic/log/tip/README?patch=&amp;linerange=314:315#3c33d39a51d3-l1.124

SuperSandro2000 commented on nixos/modules/services/web-servers/nginx/vhost-options.nix:
```diff
@@ -199,13 +199,43 @@ with lib;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable HTTP/0.9 protocol negotiation used in QUIC interoperability tests.
+        This requires using `pkgs.nginxQuic` package
+        which can be achieved by setting `services.nginx.package = pkgs.nginxQuic;`
```
Please put that into config.

Izorkin replied:
Add to the nginx.conf file?

Izorkin replied:
Added assertion.

SuperSandro2000 commented on nixos/modules/services/web-servers/nginx/vhost-options.nix:
```diff
@@ -199,13 +199,43 @@ with lib;
+        Whether to enable HTTP/0.9 protocol negotiation used in QUIC interoperability tests.
+        This requires using `pkgs.nginxQuic` package
+        which can be achieved by setting `services.nginx.package = pkgs.nginxQuic;`
+        and activate the QUIC transport protocol
+        `services.nginx.virtualHosts.<name>.quic = true;`.
```
We can assert that

Izorkin replied:
Change description?

SuperSandro2000 commented on nixos/modules/services/web-servers/nginx/vhost-options.nix:
```diff
@@ -199,13 +199,43 @@ with lib;
+      default = true;
+      description = lib.mdDoc ''
+        Whether to enable HTTP/3 protocol.
+        This requires using `pkgs.nginxQuic` package
+        which can be achieved by setting `services.nginx.package = pkgs.nginxQuic;`
+        and activate the QUIC transport protocol
+        `services.nginx.virtualHosts.<name>.quic = true;`.
```
same here

Izorkin replied:
Fixed.

SuperSandro2000 commented on nixos/modules/services/web-servers/nginx/vhost-options.nix:
```diff
@@ -188,9 +188,9 @@ with lib;
       default = true;
       description = lib.mdDoc ''
-        Whether to enable HTTP 2.
+        Whether to enable HTTP/2 protocol.
```
```suggestion
        Whether to enable the HTTP/2 protocol.
```

Izorkin replied:
Fixed.

Izorkin commented:
Update to revision 0af598651e33.

Izorkin committed:
nixos/tests/nginx: update nginx-http3 test

Izorkin commented:
Update nginx-http3 test,
